include/matx/core/allocator.h

```diff
@@ -334,6 +345,25 @@ inline void matxFree(void *ptr)
   return GetAllocMap().deallocate(ptr);
 }
 
+/**
+  Update the stream a pointer in the cache is using. This should be used when the call wants to use 
+  memory that was allocated in stream A inside of stream B. The caller must ensure that the pointer
+  and stream being used are valid.
+*/
+inline void update_stream(void *ptr, cudaStream_t stream)
+{
+  MATX_NVTX_START("", matx::MATX_NVTX_LOG_INTERNAL)
+  
+  GetAllocMap().update_stream(ptr, stream);
+}
+
+inline void matxFree(void *ptr, cudaStream_t stream)
+{
+  MATX_NVTX_START("", matx::MATX_NVTX_LOG_INTERNAL)
+  update_stream(ptr, stream);
```

Comment:
why not just take a stream into deallocate as an optional second parameter?  No need to have the update_stream API.

Reply:
since `cudaStream_t` is just a raw pointer, you cannot use a default parameter. If your default is nullptr that's the default stream, but we want a "no user-provided stream" as the option, which isn't possible since it's indistinguishable from any stream.

Reply:
i wasn't clear in what I was saying.  optional second parameter would not be a default parameter but an overload.